src/actions/textobject.ts

```diff
@@ -512,6 +512,53 @@ export class SelectInnerParagraph extends TextObjectMovement {
   }
 }
 
+@RegisterAction
+export class SelectEntire extends TextObjectMovement {
+  keys = ['a', 'e'];
+
+  public async execAction(position: Position, vimState: VimState): Promise<IMovement> {
+    let lines: number = TextEditor.getLineCount();
+    let lastLine = lines > 0 ? lines - 1 : 0;
+
+    return {
+      start: new Position(0, 0),
+      stop: new Position(lastLine, TextEditor.getLineLength(lastLine)),
+    };
+  }
+}
+
+@RegisterAction
+export class SelectEntireIgnoringLeadingTrailing extends TextObjectMovement {
+  keys = ['i', 'e'];
+
+  public async execAction(position: Position, vimState: VimState): Promise<IMovement> {
+    let lines: number = TextEditor.getLineCount();
+    let start: Position = new Position(0, 0);
+    let stop: Position = new Position(lines - 1, 0).getLineEnd();
+
+    for (let lineNo = 0; lineNo < lines; ++lineNo) {
+      let line = TextEditor.getLineAt(start);
```

Comment:
There's now `TextEditor.getLine(number)` which might be a cleaner way of doing this.

Alternatively, you could do something like

```typescript
while(start.line < lines && TextEditor.getLineAt(start).isEmptyOrWhitespace) {
  start = start.getDown()
}
```

Reply:
✔️